runtime/compiler/env/VMJ9.cpp

```diff
@@ -4811,7 +4811,7 @@ TR_J9VMBase::lookupMethodHandleThunkArchetype(uintptrj_t methodHandle)
    switch (returnType[0])
       {
       case '[':
-      case 'L':
+      case 'L': //should we add Q and with "Qjava/lang/Object;"?
```

Comment:
I'm not sure.  I would suggest that you add a case for `Q`, but keep the return type that's produced as `"Ljava/lang/Object;"` for now, and add another TODO (or maybe TBD).